olp-cpp-sdk-dataservice-read/src/repositories/PartitionsRepository.h

```diff
@@ -56,6 +58,21 @@ class PartitionsRepository final {
       client::CancellationContext cancellation_context,
       const DataRequest& data_request, client::OlpClientSettings settings);
 
+  static model::Partition PartitionFromSubQuad(
+      const model::SubQuad& sub_quad, const std::string& partition);
+
+  static PartitionsResponse QueryPartitionsAndGetDataHandle(
+      const client::HRN& catalog, const std::string& layer_id,
+      TileRequest request, int64_t version, client::CancellationContext context,
+      client::OlpClientSettings settings,
+      std::string& requested_tile_data_handle);
+
+  static model::Partitions GetTileFromCache(const client::HRN& catalog,
+                                            const std::string& layer_id,
+                                            TileRequest request,
+                                            int64_t version,
+                                            client::OlpClientSettings settings);
+
```

Comment:
Not sure I get it.
I assumed we have something like `GetVersionedTile(...)` which looks in the cache if the tiles partition is there, if not goes to the `QueryApi::QuadTreeIndex(...)` with depth 4 which downloads quad-tree and caches each subquad as partition metadata same as `PrefetchTilesRepository::GetSubQuads()`. At the end it returns the requested partition to the user.
Additional improvements can be done later on, like:
- If requested tile key not found i cache check if parent is found and if yes then assume there was a quad tree index query before but it maybe got LRU evicted in the meantime so we maybe only downlaod that specific tile in the end.
-Return nearest tile if the tile is not in the response (optional, old hype behaviour)

Reply:
1.Updated, please check. Added check if parent tile in cache. If true, get requested tile. Added test GetVersionedDataTileQuadTreeParentTileInCache.
2. Return nearest tile if the tile is not in the response. Is this applicable only for CacheOnly option?

Reply:
Those were only theoretical scenarios for the future. At least the 2. point I would omit for now as I don't see any requirement.

Reply:
Removed checking for parent tile in cache.